Source/JavaScriptCore/runtime/JSGlobalObjectFunctions.cpp

```diff
@@ -491,9 +500,9 @@ JSC_DEFINE_HOST_FUNCTION(globalFuncEval, (JSGlobalObject* globalObject, CallFram
     auto s = codeString->value(globalObject);
     RETURN_IF_EXCEPTION(scope, encodedJSValue());
 
-    if (Options::useTrustedTypes() && globalObject->requiresTrustedTypes() && !globalObject->globalObjectMethodTable()->canCompileStrings(globalObject, CompilationType::IndirectEval, s, x)) {
+    if (Options::useTrustedTypes() && globalObject->requiresTrustedTypes() && !isTrusted && !globalObject->globalObjectMethodTable()->canCompileStrings(globalObject, CompilationType::IndirectEval, s, *vm.emptyList)) {
```

Comment:
Ditto.

Reply:
Hopefully this has been addressed properly

Reply:
Ditto. This is not correct.